### ISSUE TITLE
TDLib build for MUSL libc is not ready yet

### ISSUE THREAD

**Issue by tahpot** [AlexXanderGrib/node-tdlib#13]
Thanks for a great library :)

Are you able to add support for alpine linux (we are on arm64)?

We have been using on darwin arm64, but are now deploying in docker containers.

**Comment by AlexXanderGrib** [comment]
Hello, great idea! Unfortunately i dont have a machine with arm64, it would be a great contribution if you provide build commands for tdlib on arm64

Here some resources
- https://tdlib.github.io/td/build.html
- https://github.com/AlexXanderGrib/node-tdlib/blob/main/.github/workflows/tdlib.yml#L12

If you can't, i will try my best to make this on my own, but it will definitely take longer

**Comment by tahpot** [comment]
Ok awesome, happy to help as best I can.

I have spun up an AWS instance running alpine linux in Docker.

I have tested the following script that successfully builds. Just to be clear, this build is specific to `alpine:arm64`, not any `arm64` architecture.

```bash
sudo apk update
sudo apk upgrade
sudo apk add git alpine-sdk linux-headers git zlib-dev openssl-dev gperf php cmake
git clone https://github.com/tdlib/td.git
cd td
rm -rf build
mkdir build
cd build
cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX:PATH=/usr/local ..
cmake --build . --target install
cd ..
cd ..
ls -l /usr/local
```

Here's the resulting output of `/usr/local/lib/`:

![Screenshot 2024-10-07 at 9 39 00 AM](https://github.com/user-attachments/assets/bc7a4264-80c8-4671-adf8-00c410a3ce59)

I assume you need to detect if the current machine is running alpine arm64? If so `uname -a` is probably easiest, but here's some other ideas: https://chatgpt.com/share/670316d6-d984-8001-9070-28b87e646c68

Happy to test on my end if you can create a branch that incorporates these changes.

**Comment by AlexXanderGrib** [comment]
ARM64 runners now are only available to Github Teams and Enterprise customers, [they are promising to make them available for opensource soon](https://github.com/orgs/community/discussions/19197#discussioncomment-10550689)

**Comment by tahpot** [comment]
Ah, so this is blocker. Is there any temporary work around to use something pre-built?

**Comment by AlexXanderGrib** [comment]
> Ah, so this is blocker. Is there any temporary work around to use something pre-built?

Yes, you can provide path of prebuilt tdlib to TDLibAddon.create like this

```typescript
const addon = await TDLibAddon.create('./bin/tdlib/libtdjson-arm64.so');
```

**Comment by tahpot** [comment]
I have tried building using tdlib build instructions, but receive this error:

`Telegram library error: Failed to load 'td_json_client_create': Symbol not found: node_register_module_v115`

Have you seen this before or have any ideas on how to resolve?

Update:

Does [this fix](https://github.com/eilvelia/tdl/commit/f32d94d9262b4b761de0f21e9d6ac391f722ca14) also need to be applied?

I found these related issues:

- https://github.com/tdlib/td/issues/2612
- https://github.com/eilvelia/tdl/issues/144

**Comment by AlexXanderGrib** [comment]
Fixed that, also i found a way to run alpine as linux/arm64 in docker, so i would be able to provide compiled tdlib. All this will be published in next release 2.7 as soon as i will be able to put this together. Probably tomorrow

**Comment by AlexXanderGrib** [comment]
fixed in 3.0,0

**Comment by AlexXanderGrib** [comment]
@tahpot Have you tested? May i close this?
